qt/tools/typecheck-setup.sh

```diff
@@ -6,8 +6,23 @@
 # able to resolve. A solution that doesn't require modifying the python install
 # would be welcome!
 
+set -eo pipefail
+
 TOOLS="$(cd "`dirname "$0"`"; pwd)"
-modDir=$(python -c 'import PyQt5, sys, os; print(os.path.dirname(sys.modules["PyQt5"].__file__))')
-cmd="rsync -a $TOOLS/stubs/PyQt5/* $modDir/"
+modDir=$(python -c 'import PyQt5, sys, os; sys.stdout.write(os.path.dirname(sys.modules["PyQt5"].__file__))')
+
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    CYGWIN*)
+        modDir="$(cygpath -u "${modDir}")"
+        ;;
+esac
+
+cmd="rsync -a \"${TOOLS}/stubs/PyQt5/\" \"${modDir}/\""
 
-$cmd > /dev/null 2>&1 || sudo $cmd
+if [[ "w${OS}" == "wWindows_NT" ]]; 
+then
+    eval "${cmd}" > /dev/null 2>&1 || eval "${cmd}"
```

Comment:
why the use of eval here?

Reply:
Because it is the only way to put a command in a variable with quotes on the `rsync ` arguments. For example, before the command was like this:
```shell
cmd="rsync path1 path2"
${cmd}
```

Which works fine if and only if the `path1` and `path2` do not have spaces. If there is a space, `rsync` would receive 3 arguments due argument splitting.

To fix it, quotes need to be added on the `path1` and `path2`:
```shell
cmd="rsync \"path1\" \"path2\""
${cmd}
```

However, if you run it now the shell cannot run the command because of the expansion of the quotes. You can find some examples here: https://unix.stackexchange.com/questions/23111/what-is-the-eval-command-in-bash

Reply:
I'm familiar with eval :-) It just tends to be a red flag as it can often result in security issues:

https://stackoverflow.com/questions/17529220/why-should-eval-be-avoided-in-bash-and-what-should-i-use-instead

Perhaps a cleaner way to accomplish this would be to split the command and its arguments up into separate variables instead of trying to stuff it  all into a single variable?

Also, you're running the command 3 times instead of 2 in the non-Windows case now

Reply:
(The rest of the changes look good, thanks. And a Windows CI would be nice to have as well, thank you for working on that)

Reply:
```sh
# I tried this
cmd="-a '${TOOLS}/stubs/PyQt5/' '${modDir}/'"
rsync ${cmd}

# And this:
cmd="-a \"${TOOLS}/stubs/PyQt5/\" \"${modDir}/\""
rsync ${cmd}

# But in both cases if fails with the error as originally, i.e.,
# bash still requires the eval thing to evaluate the quotes inside the variable.
+ rsync -a '"${TOOLS}/stubs/PyQt5/"' '"${modDir}/"'
+ rsync -a '"${TOOLS}/stubs/PyQt5/"' '"${modDir}/"'
rsync: change_dir "/cygdrive/f/anki/qt//"${TOOLS}/stubs/PyQt5" failed: No such file or directory (2)
rsync: change_dir#3 "/cygdrive/f/anki/qt//"${modDir}" failed: No such file or directory (2)
rsync error: errors selecting input/output files, dirs (code 3) at main.c(713) [Receiver=3.1.2]

+ rsync -a ''\''/cygdrive/f/anki/qt/tools/stubs/PyQt5/'\''' ''\''/cygdrive/f/anki/pyenv/lib/site-packages/PyQt5/'\'''
+ rsync -a ''\''/cygdrive/f/anki/qt/tools/stubs/PyQt5/'\''' ''\''/cygdrive/f/anki/pyenv/lib/site-packages/PyQt5/'\'''
rsync: change_dir "/cygdrive/f/anki/qt//'/cygdrive/f/anki/qt/tools/stubs/PyQt5" failed: No such file or directory (2)
rsync: change_dir#3 "/cygdrive/f/anki/qt//'/cygdrive/f/anki/pyenv/lib/site-packages/PyQt5" failed: No such file or directory (2)
rsync error: errors selecting input/output files, dirs (code 3) at main.c(713) [Receiver=3.1.2]
```

> Also, you're running the command 3 times instead of 2 in the non-Windows case now

I was rerunning the command because the first time its `stdout` and `stderr` were going to `/dev/null`, then, on error I rerun a second time to log out what error was happening before going into sudo mode.

I just pushed this, removing the variable and duplicating the command:
```sh
if [[ "w${OS}" == "wWindows_NT" ]];
then
    rsync -a "${TOOLS}/stubs/PyQt5/" "${modDir}/"
else
    rsync -a "${TOOLS}/stubs/PyQt5/" "${modDir}/" || sudo rsync -a "${TOOLS}/stubs/PyQt5/" "${modDir}/"
fi
```

I removed the `/dev/null` thing because here `rsync` was already not outputting anything to `stderr/stdout`. `rsync` would just be quite verbose if we were using the `-v` flag. I just tested here both on Windows and Linux and rsync is not outputting anything when there are no errors:
![image](https://user-images.githubusercontent.com/5332158/77367591-3f40db80-6d39-11ea-8a74-28267c069a84.png)

So, removing the `/dev/null` speeds things up when there is some error on `rsync` command, i.e., I would not have to push or edit the file removing it to figure out why `rsync` would not be working in case of some error.